diff/diff.go

```diff
@@ -24,7 +23,11 @@ func diffBase(mainBranch bool, baseBranch string) (string, error) {
 	return strings.TrimRight(string(mergeBase), "\n"), nil
 }
 
-func changedFiles(mainBranch bool, baseBranch string) ([]string, error) {
+// ChangedFiles uses git to determine the list of files that changed for
+// the current revision.
+// It can operate in a normal (branch) mode, where it compares to a 'baseBranch',
+// or a 'mainBranch' mode, where it compares to the previous revision
+func ChangedFiles(mainBranch bool, baseBranch string) ([]string, error) {
```

Comment:
Whenever you run an `exec.Command`, it might be useful to attach to `stderr` so that errors are captured:
```go
	gitMergeBase := exec.Command("git", "merge-base", baseBranch, "HEAD")
	gitMergeBase.Stderr = os.Stderr
	mergeBase, err := gitMergeBase.Output()
```

Reply:
the `Output()` method does that and we later print the error it returns, I think

Reply:
nice